async-http-client-backend/zio/src/test/scala/sttp/client/asynchttpclient/zio/WebSocketStubZioTests.scala

```diff
@@ -0,0 +1,119 @@
+package sttp.client.asynchttpclient.zio
+
+import org.scalatest.concurrent.ScalaFutures
+import sttp.client._
+import sttp.client.testing.WebSocketStub
+import sttp.client.impl.zio._
+import sttp.model.Headers
+import sttp.model.ws.WebSocketFrame
+import zio._
+import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatest.matchers.should.Matchers
+import sttp.client.ws.WebSocketEvent
+import scala.util.Success
+import scala.util.Failure
+
+class WebSocketStubZioTests extends AnyFlatSpec with Matchers with ScalaFutures {
+
+  def mkBackend(webSocketStub: WebSocketStub[_]) =
+    AsyncHttpClientZioBackend.stub
+      .whenRequestMatches(_ => true)
+      .thenRespondWebSocket(Headers(List.empty), webSocketStub)
+
+  "web socket stub" should "return initial Incoming frames on 'receive'" in {
+    val frames = List("a", "b", "c").map(WebSocketFrame.text(_))
+    val webSocketStub = WebSocketStub.withIncomingFrames(frames)
+    implicit val b = mkBackend(webSocketStub)
+    val test = for {
+      handler <- ZioWebSocketHandler()
+      ws <- basicRequest.get(uri"http://example.org/a/b/c").openWebsocket(handler).map(_.result)
+      msg1 <- ws.receive
+      msg2 <- ws.receive
+      msg3 <- ws.receive
+    } yield List(msg1, msg2, msg3)
+
+    runtime.unsafeRun(test) shouldBe frames.map(Right(_))
+  }
+
+  it should "return initial responses on 'receive'" in {
+    val okFrame = WebSocketFrame.text("abc")
+    val exception = new Exception("boom")
+    val webSocketStub = WebSocketStub.withMessages(List(Success(Right(okFrame)), Failure(exception)))
+    implicit val b = mkBackend(webSocketStub)
+    val test = for {
+      handler <- ZioWebSocketHandler()
+      ws <- basicRequest.get(uri"http://example.org/a/b/c").openWebsocket(handler).map(_.result)
+      msg <- ws.receive
+      err <- ws.receive.either
+    } yield (msg, err)
+
+    runtime.unsafeRun(test) shouldBe ((Right(okFrame), Left(exception)))
+  }
+
+  it should "add next Incoming frames as reaction to send" in {
+    val firstFrame = WebSocketFrame.text("No. 1")
+    val secondFrame = WebSocketFrame.text("Second")
+    val thridFrame = WebSocketFrame.text("3")
```

Comment:
typo: `thrid`

Reply:
Thanks! My dyslexia is very well. I had to read it 3 times before I've finally seen it's `thrid`.